ega-data-api-htsget/src/main/java/eu/elixir/ega/ebi/htsget/service/internal/TicketServiceV2Impl.java

```diff
@@ -0,0 +1,203 @@
+                                    Optional<Long> start,
+                                    Optional<Long> end,
+                                    Optional<List<Field>> fields,
+                                    Optional<List<String>> tags,
+                                    Optional<List<String>> notags)
+            throws HtsgetException, NotFoundException, PermissionDeniedException, IOException, URISyntaxException {
+
+        boolean onlyHeader = false;
+        if (requestClass.isPresent()) {
+            if (!requestClass.get().equalsIgnoreCase("Header"))
+                throw new InvalidInputException("Invalid class parameter");
+            onlyHeader = true;
+            if (referenceName.isPresent() ||
+                    start.isPresent() ||
+                    end.isPresent() ||
+                    fields.isPresent() ||
+                    tags.isPresent() ||
+                    notags.isPresent())
+                throw new InvalidInputException("Other parameters not allow for Header request");
+        }
+
+        if (start.isPresent()) {
+            if (!referenceName.isPresent())
+                throw new InvalidInputException("Must specify reference name if using start parameter");
+
+            if (referenceName.get().equals("*"))
+                throw new InvalidInputException("Cannot use start with unplaced unmapped reads");
+        }
+
+        // Ascertain Access Permissions for specified File ID
+        File reqFile = fileInfoService.getFileInfo(id);
+
+        // Start Building the URL for the file on DataEdge
+        URI baseURI = UriComponentsBuilder
+                .fromUriString(externalConfig.getEgaExternalUrl() + id)
+                .queryParam("destinationFormat", "plain")
+                .build().toUri();
+
+        HtsgetResponseV2 result = new HtsgetResponseV2(format);
+
+        if (!onlyHeader && !referenceName.isPresent()) {
+            // user is requesting the entire file
+            result.addUrl(new HtsgetUrlV2(baseURI));
+            if (reqFile.getUnencryptedChecksumType() != null && reqFile.getUnencryptedChecksumType().equalsIgnoreCase("md5"))
+                result.setMd5(reqFile.getUnencryptedChecksum());
+
+        } else {
+
+            DataProvider reader = dataProviderFactory.getProviderForFormat(format);
+            if (!reader.supportsFileType(reqFile.getFileName()))
+                throw new UnsupportedFormatException("Conversion not supported");
+
+            try (SeekableStream dataStream = resClient.getStreamForFile(id)) {
+                reader.readHeader(dataStream);
+                result.addUrl(new HtsgetUrlV2(reader.getHeaderAsDataUri(), "header"));
+
+                if (!onlyHeader) {
+
+                    if (referenceName.get().equals("*")) {
+                        // TODO
+                        throw new NotImplementedException("Unplaced unmapped reads not implemented yet");
+                    }
```

Comment:
Substitute for an assert.

Reply:
I don't understand, do you mean I should extract this to a separate method?

Reply:
Yup same as the other

Reply:
So, was this `NotImplementedException("Unplaced unmapped reads not implemented yet")` at line 102 substituted with an assert in the end?